src/Features/Core/Portable/ExternalAccess/UnitTesting/API/INewUnitTestingIncrementalAnalyzerImplementation.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
+{
+    internal interface INewUnitTestingIncrementalAnalyzerImplementation
+    {
+#if false // Not used in unit testing crawling
+        // Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancellationToken);
+        // Task DocumentOpenAsync(Document document, CancellationToken cancellationToken);
+        // Task DocumentCloseAsync(Document document, CancellationToken cancellationToken);
+        // Task DocumentResetAsync(Document document, CancellationToken cancellationToken);
+        // void RemoveProject(ProjectId projectId);
```

Comment:
There is an assumption on our end that we will always be told about removed documents in removed projects via RemoveDocument() below. As we remove the associated code behind RemoveProject (and comment out other parts of the crawler implementation), it would be great to ensure that we won't inadvertently break assumptions like the above one.

Will existing tests for the crawler being ported to EA as well? Can you add some tests for this?

Reply:
yes.  my expectation is that we will port all *viable* tests over to this new crawler.  e.g. all tesst that test the functionality that we're preserving.  any functionality that we're removing we'll not keep tests for.